src/components/VerticalNavigation/VerticalNavigation.js

```diff
@@ -0,0 +1,482 @@
+    }
+    onMenuToggleClick && onMenuToggleClick();
+  }
+
+  updateNavOnItemHover(primary, secondary, tertiary) {
+    const { onItemHover } = this.props;
+    this.updateHoverState(true, primary, secondary, tertiary);
+    if (onItemHover) {
+      onItemHover(primary, secondary, tertiary);
+    }
+  }
+
+  updateNavOnItemBlur(primary, secondary, tertiary) {
+    const { onItemBlur, setControlledState } = this.props;
+    const { numHoveredPrimary, numHoveredSecondary } = this.updateHoverState(
+      false,
+      primary,
+      secondary,
+      tertiary
+    );
+    if (primary && !secondary && !tertiary && numHoveredPrimary === 0) {
+      setControlledState({ hoverSecondaryNav: false });
+    }
+    if (primary && secondary && !tertiary && numHoveredSecondary === 0) {
+      setControlledState({ hoverTertiaryNav: false });
+    }
+    if (onItemBlur) {
+      onItemBlur(primary, secondary, tertiary);
+    }
+  }
+
+  updateNavOnBlurAll() {
+    const { setControlledState } = this.props;
+    setControlledState({ hoverSecondaryNav: false, hoverTertiaryNav: false });
+  }
+
+  updateNavOnItemClick(primary, secondary, tertiary) {
+    const { onItemClick, isMobile } = this.props;
+    const item = deepestOf(primary, secondary, tertiary);
+    if (isMobile) {
+      this.setMobilePath(null);
+    }
+    if (!isMobile || !item.subItems || item.subItems.length === 0) {
+      this.navigateToItem(item);
+    }
+    onItemClick && onItemClick(primary, secondary, tertiary);
+  }
+
+  updateNavOnPinSecondary(pinned) {
+    const { onPinSecondary, isMobile, setControlledState } = this.props;
+    if (!isMobile) {
+      setControlledState({ pinnedSecondaryNav: pinned });
+      onPinSecondary && onPinSecondary(pinned);
+    }
+    setControlledState({ hoverSecondaryNav: false });
+  }
+
+  updateNavOnPinTertiary(pinned) {
+    const { onPinTertiary, isMobile, setControlledState } = this.props;
+    if (!isMobile) {
+      setControlledState({ pinnedTertiaryNav: pinned });
+      onPinTertiary && onPinTertiary(pinned);
+    }
+    setControlledState({ hoverSecondaryNav: false, hoverTertiaryNav: false });
+    if (pinned) {
+      this.updateNavOnPinSecondary(false);
+    }
+  }
+
+  updateNavOnMobileSelection(primary, secondary, tertiary) {
+    const { onMobileSelection } = this.props;
+    // All the behavior here is handled by mobilePath and setMobilePath,
+    // but we still make a callback available here.
+    onMobileSelection && onMobileSelection(primary, secondary, tertiary);
+  }
+
+  setActivePath(activePath) {
+    this.props.setControlledState({ activePath });
+  }
+
+  setMobilePath(mobilePath) {
+    this.props.setControlledState({ mobilePath });
+  }
+
+  forceHideSecondaryMenu() {
+    // TODO do we really need this?
+    this.setState({ forceHidden: true });
+    setTimeout(() => {
+      this.setState({ forceHidden: false });
+    }, 500);
+  }
+
+  navigateToItem(item) {
+    console.log('TODO NAVIGATE!', item); // TODO
```

Comment:
should this just call a callback function if one exists for now? fine with leaving it open ended for v1 if others are...

Reply:
Yeah, i forgot I hadn't done that already. I'll fill it in with an onNavigate callback you can use, but the item components themselves also have onClick and href props that can be used instead.